### ISSUE TITLE
> @Cernelius please raise such create a bug report when you notice such problems. The less time passing between a problem being introduced and noticed tends to reduce the cost for a fix.

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#7825]
> @Cernelius please raise such create a bug report when you notice such problems. The less time passing between a problem being introduced and noticed tends to reduce the cost for a fix.

The problem is that TripleA has so many problems and wrong behaviours that, if I would report everything I know, that would be really a massive amount of issues, so what I do is making a selection myself and trying to focus on what I believe is most important (and, even in that case, the majority of times I report something the problem is not fixed). So, while you are correct in theory, in practice the matter for me is to find a balance between the effort of documenting and discussing the issues and my perceived chance at seeing it actually fixed (I would just never report anything if I would believe that such chance is 0%).

Aside from this, there is a chance that at least some of the issues I raised in here I've already reported, maybe in one or more closed issues (I cannot remember everything).

_Originally posted by @Cernelius in https://github.com/triplea-game/triplea/pull/7823#issuecomment-703056645_

**Comment by DanVanAtta** [comment]
Closing, will leave to the original Author to move this to a dedicated issue.

**Comment by DanVanAtta** [comment]
> and, even in that case, the majority of times I report something the problem is not fixed

Some of that is lack of labor, some of that is certainly up-front cost @Cernelius . If a problem is prefaced with 30 minutes of reading, then nothing is going to get done. For example, I had 2 hours tonight, I've spent 1 hour in the forums, 30 minutes reviewing code, and now 30 minutes just responding to this topic and thinking about the problems raised. That is 2 hours not spent coding (in an ideal world, it'll never be the case that 0 time is for communication). The point is that the closer to a technical spec that we have that is short and concise, the more fixes you'll see. If I can read something in 5 minutes, know exactly what the problem is, what the fix should be, and have a save, then I can dive right  into where the the problem and start translating the problem statement into which code is at fault and translating the solution statement into a new code that needs to be created. Spending time simply understanding what the problem is, demonstrating it, getting to that save, checking for how to solve it, is time not spent coding and just adds to the total time to fix. 

In short, part of the problem is that the time to fix is inefficient here. The cost of a fix is exponential to the length and complexity of the bug report. A 10 page bug report for example could take 100 hours to fix, meanwhile if that were broken down into 10, 1 page bug reports, it could then take 10 hours to fix the same.

> if I would report everything I know, that would be really a massive amount of issues,

You raise a good point here. I think there is a failure of tracking. If we had a lot of high quality bug reports that we could know were valid and could categorize easily so that it would not be costly to even ask "what should I work on next", then we might make better progress on this.

**Comment by DanVanAtta** [comment]
One idea, if we agree we want to focus on one topic, we can use projects and the task cards in projects for tracking many issues. Perhaps instead of labels, we could use projects instead for different areas of focus.

**Comment by DanVanAtta** [comment]
Fundamentally, at the end of the day, @Cernelius you're right that most issues are not getting fixed. I don't know if we'll ever solve it. The problem is specifically the number of incoming problems is greater than us fixing them. To fix problems faster, we need deep changes to the code to untie the knots and that requires massive time. On the other side, we need to spend less time understanding the incoming problems. If we accept that, then we can understand that the problems that take a lot of time to simply read and understand are a bad use of time when instead we could be fixing 3 or 4 problems where the bug reporter carefully crafted an exact 50 words to describe one problem and what should be done to fix it and attached a save game at the point of error. If we were to go draconian on that route, any bug  report over 300 words wand without a save game we would just summarily close (cause we understand we'll never fix it, so spend the time where it's efficient).

Perhaps on issues that are lingering, what you can do @Cernelius is ask in the issue if the problem and fix are clear. Alternatively, make sure every issue has a two sentence, "here is what needs to happen next for this to be solved". We need to remove the 30 minutes of reading 10 pages to understand just where a problem is so we can instead spend that time on fixing the problem. 

Said another way, I don't think the problem is just we're getting too many issues. I think the problem is we're getting too many fuzzily defined issues with too much conversation. My experience with dealing bug queues, the answer to that is you consistently redefine bugs into multiple smaller well defined bugs each one being more and more concise. In which case all issues take on one of two flavors: concise bug report, or B) conversation that results in concise bug reports. You are then left at the end of day never needing to read more than one paragraph of text to then jump in and start working on a fix.
